xonsh/aliases.py

```diff
@@ -550,6 +550,9 @@ def source_foreign_fn(
 source_foreign = ArgParserAlias(
     func=source_foreign_fn, has_args=True, prog="source-foreign"
 )
+source_foreign.parser.add_argument(
```

Comment:
Though it is correct, adding it here would cause the parser to be built during import time. Instead we can update the parameters.

Reply:
Is there currently a way of specifying multiple parameter sets for a single variable?  Otherwise I can make it lazy

Reply:
Yes , the ArgParserAlias can be sub-classed. It will work with the other composed functions as well.

Reply:
here is the change  that would do it and work for source-bash/zsh 
https://github.com/xonsh/xonsh/commit/36b05d3d74afd09953dbfdadf378ae39b82f289e

Reply:
https://github.com/xonsh/xonsh/compare/main...jnoortheen:xonsh:update_source_fn_interactive